robot/impl/resource_manager_modular_test.go

```diff
@@ -388,6 +388,10 @@ type dummyModMan struct {
 	state      map[resource.Name]bool
 }
 
+func (m *dummyModMan) Add(ctx context.Context, confs ...config.Module) error {
+	return errors.New("not implemented")
+}
```

Comment:
required due to https://github.com/viamrobotics/rdk/pull/3641/files#r1518297402

Reply:
If you guard the `Add` call in `updateResources` with `if conf.Added.Modules != nil` I think you won't need this.

Reply:
ah yeah you're right - I was relying on `nil...` expanding to an empty array, which would be a noop for `Add`, but I'll add the `nil` check to preserve the original behavior a bit more closely.